### PR TITLE
Added external quarkus_openshift test

### PR DESCRIPTION
**Pull request by LongyuZhang**
- enabled external quarkus_openshift test, it remains disabled due to lack of openshift cluster
- added arch detection to select corresponding openshift cluster
- added platform requirement and subsets

runtimes/backlog 353

Signed-off-by: Longyu Zhang <longyu.zhang@ibm.com>

### CODE REVIEW

**Comment by LongyuZhang** [comment]
Grinder links:
x86: grinder_internal/10080
s390x: grinder_internal/10067
ppc64le: grinder_internal/10068

**Comment by LongyuZhang** [comment]
@smlambert @llxia Could you review it? Thanks.

File: external/quarkus_openshift/run_test.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+cd $DEST
+pwd
+
+git clone https://github.com/quarkus-qe/quarkus-openshift-test-suite.git
+cd quarkus-openshift-test-suite
+
+echo "Compile and run quarkus_openshift tests"
+
+if [[ $ARCH =~ "x86" ]]; then
+    oc login --token=bgn5pEjb_1mnyDfAtj6hwOAfRL1WBpLXb6R-C8zsAdQ --server=https://api.digger.cp.fyre.ibm.com:6443 --insecure-skip-tls-verify
```
Comment (smlambert) [inline external/quarkus_openshift/run_test.sh:30]: We will have to have a way to parameterize the token and server values (perhaps using environment variables set to the value of whatever the user who wants to run the test requires).

File: external/quarkus_openshift/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    ./mvnw -pl \!configmap/file-system,\!config-secret/file-system,\!messaging/artemis,\!messaging/artemis-jta,\!messaging/amqp-reactive,\!messaging/qpid,\!sql-db/postgresql,\!sql-db/mssql,\!security/keycloak,\!security/keycloak-authz,\!microprofile,\!external-applications/todo-demo-app,\!deployment-strategies/quarkus clean verify -Dquarkus.s2i.base-jvm-image=docker.io/cosmoxiix/adopt11openj9s390x -Dversion.quarkus=1.7.2.Final -Dts.use-ephemeral-namespaces
+elif [[ $ARCH =~ "ppc64le" ]]; then
+    oc login --token=XDFAomeetJyZXlJVgWRxlZX2QsFGIVTfmvFGwQricSE --server=https://api.ocple.cp.fyre.ibm.com:6443 --insecure-skip-tls-verify
+    ./mvnw -pl \!configmap/file-system,\!config-secret/file-system,\!messaging/artemis,\!messaging/artemis-jta,\!messaging/amqp-reactive,\!messaging/qpid,\!sql-db/mssql,\!security/keycloak,\!security/keycloak-authz,\!microprofile,\!deployment-strategies/quarkus clean verify -Dquarkus.s2i.base-jvm-image=docker.io/cosmoxiix/adopt11openj9px -Dversion.quarkus=1.7.2.Final -Dts.use-ephemeral-namespaces
+else
+    echo "This arch is no supported."
```
Comment (smlambert) [inline external/quarkus_openshift/run_test.sh:39]: Include the value of the ARCH in the echo "$ARCH arch is not supported"

**Review by smlambert — CHANGES_REQUESTED** [review]
Thanks @LongyuZhang !  Can you try changing to use env vars for token/server values, to allow those to be flexibly set as part of run_test.sh?

File: external/quarkus_openshift/run_test.sh (hunk 1 above)
Comment (LongyuZhang) [reply external/quarkus_openshift/run_test.sh]: Thanks for the comment, I have changed ``` OCP_SERVER``` and```OCP_TOKEN``` into parameters, and tested in Grinder_TKG through adding these two parameters with @llxia 's help. The new test links are: Grinder_TKG 776, 777, and 779. Thanks.

File: external/quarkus_openshift/run_test.sh (hunk 2 above)
Comment (LongyuZhang) [reply external/quarkus_openshift/run_test.sh]: The echo message has been updated as well. Thanks.

**Review by smlambert — APPROVED** [review]
thanks @LongyuZhang !

**Review by llxia — APPROVED** [review]
LGTM

**Comment by smlambert** [comment]
Summary from Longyu of the steps to get quarkus-openshift-test running:

- Used Openshift oc command line tool to communicate with Openshift Cluster, which runs on corresponding platform, e.g. Jenkins x86 machine will connect with Openshift Cluster runs on x86 platform. Possible improvement is to install oc tools on all Jenkins machines, currently only on machines with label sw.tool.oc;
- modified [s2i source code](https://github.com/fabric8io-images/s2i), which is used by Redhat to generate s2i images, to generate adopt-11-openj9 s2i images for [x,p](https://hub.docker.com/repository/registry-1.docker.io/cosmoxiix/adopt11openj9px/tags?page=1),[z](https://hub.docker.com/r/cosmoxiix/adopt11openj9s390x/tags) platform, which were pushed to my personal DockerHub and are used when running quarkus-openshift-test-suite. Possible improvement is to generate adopt 11 s2i nightly image, same as [adopt 11 docker images](https://hub.docker.com/r/adoptopenjdk/openjdk11/);
- the minimum quarkus version in test should be at least 1.7.2.Final, since the io.quarkus:quarkus-maven-plugin issue on plinux and zlinux machines only gets fixed after opening the issue previously;

**Pull request merged**